src/test/java/com/google/api/gax/grpc/UnaryCallableTest.java

```diff
@@ -458,25 +460,72 @@ public Integer extractPageSize(Integer payload) {
     }
   }
 
-  private class PagedListResponse extends PagedListResponseImpl<Integer, List<Integer>, Integer> {
-    public PagedListResponse(
-        UnaryCallable<Integer, List<Integer>> callable,
-        PagedListDescriptor<Integer, List<Integer>, Integer> pageDescriptor,
-        Integer request,
-        CallContext context) {
-      super(callable, pageDescriptor, request, context);
+  private static class ListIntegersPagedResponse
+      extends AbstractPagedListResponse<
+          Integer, List<Integer>, Integer, ListIntegersPage, ListIntegersSizedPage> {
+
+    protected ListIntegersPagedResponse(ListIntegersPage page) {
+      super(page, ListIntegersSizedPage.createEmptyCollection());
+    }
+
+    public static ListIntegersPagedResponse create(
+        PageContext<Integer, List<Integer>, Integer> context, List<Integer> response) {
+      ListIntegersPage page = new ListIntegersPage(context, response);
+      return new ListIntegersPagedResponse(page);
+    }
+  }
+
+  private static class ListIntegersPage
+      extends AbstractPage<Integer, List<Integer>, Integer, ListIntegersPage> {
+
+    public ListIntegersPage(
+        PageContext<Integer, List<Integer>, Integer> context, List<Integer> response) {
+      super(context, response);
+    }
+
+    @Override
+    protected ListIntegersPage createPage(
+        PageContext<Integer, List<Integer>, Integer> context, List<Integer> response) {
+      return new ListIntegersPage(context, response);
+    }
+  }
+
+  private static class ListIntegersSizedPage
+      extends AbstractFixedSizeCollection<
+          Integer, List<Integer>, Integer, ListIntegersPage, ListIntegersSizedPage> {
+
+    private ListIntegersSizedPage(List<ListIntegersPage> pages, int collectionSize) {
+      super(pages, collectionSize);
+    }
+
+    private ListIntegersSizedPage() {
+      super();
+    }
+
+    private static ListIntegersSizedPage createEmptyCollection() {
+      return new ListIntegersSizedPage();
```

Comment:
It doesn't feel like the empty constructor is really necessary - couldn't we have this call super(null, 0), and relax the constraints in AbstractFixedSizeCollection to allow it?